### PR TITLE
Add Unison version 2.48

### PR DESCRIPTION
**Pull request by fasmat**
### Reasoning:

Unsion versions are not compatible with each other. If you want to synchronize e.g. an Ubuntu system with OS X, you need depending on the ubuntu version either 2.40 or 2.48. See here: https://packages.ubuntu.com/search?keywords=unison

The same would be true if you want to synchronize data between different OS X versions, where the homebrew-cask installs a different version based on the version of the OS.

### Boilerplate:

After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, if **adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not [already refused].
- [x] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/Homebrew/homebrew-cask-versions/pulls
[already refused]: https://github.com/Homebrew/homebrew-cask-versions/search?q=is%3Aclosed&type=Issues
[the correct repo]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

### CODE REVIEW

**Comment by fasmat** [comment]
Any chance this will get merged?

File: Casks/unison248.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  # github.com/bcpierce00/unison was verified as official when first introduced to the cask
+  url "https://github.com/bcpierce00/unison/releases/download/#{version}/Unison-OS-X-#{version}.zip"
+  appcast 'https://github.com/bcpierce00/unison/releases.atom'
+  name 'Unison'
+  homepage 'https://www.cis.upenn.edu/~bcpierce/unison/'
+
+  conflicts_with formula: 'unison',
+                 cask:    [
+                            'unison',
+                            'unison240',
```
Comment (vitorgalvao) [inline Casks/unison248.rb]: What are these? They don’t seem to exist. Are you making them in anticipation to this being merged?

**Comment by vitorgalvao** [comment]
https://github.com/Homebrew/homebrew-cask-versions/pull/6500#issuecomment-439421098

**Comment by fasmat** [comment]
Created a tap for anyone interested: https://github.com/fasmat/homebrew-unison

**Pull request closed without merging**